pkg/ipam/node_manager.go

```diff
@@ -181,9 +181,12 @@ func NewNodeManager(instancesAPI AllocationImplementation, k8sAPI CiliumNodeGett
 
 // Start kicks of the NodeManager by performing the initial state
 // synchronization and starting the background sync go routine
-func (n *NodeManager) Start(ctx context.Context) {
+func (n *NodeManager) Start(ctx context.Context) error {
 	// Trigger the initial resync in a blocking manner
-	n.instancesAPI.Resync(ctx)
+	resyncTime := n.instancesAPI.Resync(ctx)
+	if resyncTime.IsZero() {
+		return fmt.Errorf("Initial synchronization with instances API failed")
```

Comment:
Nit: we can use `errors.New()` for simple string errors. This can help reduce the usage and hopefully the need to import `fmt`, but I assume that we already use `fmt.Errorf` for good reasons in this file. Feel free to ignore if so.